### ISSUE TITLE
Debug build over CUDA 10.1 always fails

### ISSUE THREAD

**Issue by xuhdev** [pytorch/pytorch#34079]
When building with CUDA 10.1, gcc 8.3 on Linux, I always got the following error message:

```
ptxas /tmp/hong/tmpxft_0000359d_00000000-5_SparseCUDATensorMath.ptx, line 49217; error   : Ali
gnment of argument does not match formal parameter '_'                                        
ptxas /tmp/hong/tmpxft_0000359d_00000000-5_SparseCUDATensorMath.ptx, line 49270; error   : Ali
gnment of argument does not match formal parameter '_'                                        
ptxas fatal   : Ptx assembly aborted due to errors                                            
CMake Error at torch_cuda_generated_SparseCUDATensorMath.cu.o.Debug.cmake:281 (message):      
  Error generating file                                                                       
  /home/hong/wsrc/pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/spars
e/cuda/./torch_cuda_generated_SparseCUDATensorMath.cu.o   

```
## To Reproduce

1. `DEBUG=1 ONNX_ML=0 MAX_JOBS=8 CMAKE_CXX_COMPILER_LAUNCHER=ccache CMAKE_C_COMPILER_LAUNCHER=ccache CMAKE_CUDA_COMPILER_LAUNCHER=ccache USE_OPENMP=0 USE_DISTRIBUTED=0 USE_MKLDNN=0 USE_NCCL=0 USE_CUDA=1 USE_CUDNN=0 USE_STATIC_CUDNN=0 USE_NNPACK=0 USE_QNNPACK=0 USE_FBGEMM=0 BUILD_TEST=0 TORCH_CUDA_ARCH_LIST="6.1" python setup.py develop --cmake-only`
1. `python setup.py develop`

## Environment

 - PyTorch Version (e.g., 1.0): master
 - OS (e.g., Linux): Linux
 - Build command you used (if compiling from source): `DEBUG=1 ONNX_ML=0 MAX_JOBS=8 CMAKE_CXX_COMPILER_LAUNCHER=ccache CMAKE_C_COMPILER_LAUNCHER=ccache CMAKE_CUDA_COMPILER_LAUNCHER=ccache USE_OPENMP=0 USE_DISTRIBUTED=0 USE_MKLDNN=0 USE_NCCL=0 USE_CUDA=1 USE_CUDNN=0 USE_STATIC_CUDNN=0 USE_NNPACK=0 USE_QNNPACK=0 USE_FBGEMM=0 BUILD_TEST=0 TORCH_CUDA_ARCH_LIST="6.1" python setup.py develop --cmake-only`
 - Python version: 3.6
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: P1000

cc @ngimel

**Comment by zasdfgbnm** [comment]
The behavior of DEBUG build is changed in https://github.com/pytorch/pytorch/commit/c66ca74f03b6ff090142c34b602b5a4c45523d47 which might be related. (`REL_WITH_DEB_INFO` behavior was also changed in the same PR, but `REL_WITH_DEB_INFO` is working pretty well). And that error message looks like a compiler bug to me.

**Comment by zasdfgbnm** [comment]
But for now, let me try if replacing the debug flag ` -g -G --source-in-ptx` with ` -g -lineinfo --source-in-ptx` could solve the problem

**Comment by zasdfgbnm** [comment]
Quote from NVCC document:

```
4.2.3.2. --debug (-g)
Generate debug information for host code.

4.2.3.3. --device-debug (-G)
Generate debug information for device code.

This option turns off all optimizations on device code. It is not intended for profiling; use --generate-line-info instead for profiling.

4.2.3.5. --generate-line-info (-lineinfo)
Generate line-number information for device code.

4.2.8.2. --source-in-ptx (-src-in-ptx)
Interleave source in PTX.

May only be used in conjunction with --device-debug or --generate-line-info.
```

**Comment by xuhdev** [comment]
Solved by #34153
